sjsonnet/src/sjsonnet/Evaluator.scala

```diff
@@ -30,57 +30,63 @@ class Evaluator(resolver: CachedResolver,
   var tailstrict: Boolean = false
 
   override def visitExpr(e: Expr)(implicit scope: ValScope): Val = try {
-    e match {
-      case e: ValidId => visitValidId(e)
-      case e: BinaryOp => visitBinaryOp(e)
-      case e: Select => visitSelect(e)
-      case e: Val => e
-      case e: ApplyBuiltin1 => visitApplyBuiltin1(e)
-      case e: ApplyBuiltin2 => visitApplyBuiltin2(e)
-      case e: ApplyBuiltin3 => visitApplyBuiltin3(e)
-      case e: ApplyBuiltin4 => visitApplyBuiltin4(e)
-      case e: And => visitAnd(e)
-      case e: Or => visitOr(e)
-      case e: UnaryOp => visitUnaryOp(e)
-      case e: Apply1 => visitApply1(e)
-      case e: Lookup => visitLookup(e)
-      case e: Function => visitMethod(e.body, e.params, e.pos)
-      case e: LocalExpr => visitLocalExpr(e)
-      case e: Apply => visitApply(e)
-      case e: IfElse => visitIfElse(e)
-      case e: Apply3 => visitApply3(e)
-      case e: ObjBody.MemberList => visitMemberList(e.pos, e, null)
-      case e: Apply2 => visitApply2(e)
-      case e: AssertExpr => visitAssert(e)
-      case e: ApplyBuiltin => visitApplyBuiltin(e)
-      case e: Comp => visitComp(e)
-      case e: Arr => visitArr(e)
-      case e: SelectSuper => visitSelectSuper(e)
-      case e: LookupSuper => visitLookupSuper(e)
-      case e: InSuper => visitInSuper(e)
-      case e: ObjExtend => visitObjExtend(e)
-      case e: ObjBody.ObjComp => visitObjComp(e, null)
-      case e: Slice => visitSlice(e)
-      case e: Import => visitImport(e)
-      case e: Apply0 => visitApply0(e)
-      case e: ImportStr => visitImportStr(e)
-      case e: ImportBin => visitImportBin(e)
-      case e: Expr.Error => visitError(e)
-      case e => visitInvalid(e)
+    (e._tag: @switch) match {
```

Comment:
As Expr is not a class, then `e._tag` is actually an `invokeinterface`, it would be better to make it a `getField`.